### PR TITLE
Support multiple sheet instances

### PR DESCRIPTION
**Pull request by AndrewPrifer**
This PR adds support for multiple sheet instances of the same sheet, enabling creating editable components backed by the same theatre sheet.

Aside from that, the PR also slightly modifies the SheetProvider API, changing the getSheet prop to simply sheet, allowing passing in the sheet itself directly, while also ensuring that the sheet context always provides a value from the very first render.

### CODE REVIEW

File: packages/r3f/src/utils.ts [hunk 1]
```diff
@@ -1,3 +1,7 @@
 import {useEditorStore} from './store'
+import type {ISheet} from '@theatre/core'
 
 export const refreshSnapshot = useEditorStore.getState().createSnapshot
+
+export const makeObjectKey = (sheet: ISheet, name: string) =>
+  `${sheet.address.sheetId}:${sheet.address.sheetInstanceId}:${name}`
```
Comment (AriaMinaei) [inline packages/r3f/src/utils.ts:7]: I was confused about the word "objectKey" as that's also the word we use in `@theatre/core` for `sheet.object(key, ...)`. Maybe rename it to `makeStoreKey()`?

File: packages/r3f/src/components/editable.tsx [hunk 2]
```diff
@@ -47,7 +48,9 @@ const createEditable = <Keys extends keyof JSX.IntrinsicElements>(
 
         const objectRef = useRef<JSX.IntrinsicElements[U]>()
 
-        const sheet = useCurrentSheet()
+        const sheet = useCurrentSheet()!
+
+        const objectKey = makeObjectKey(sheet, uniqueName)
```
Comment (AriaMinaei) [inline packages/r3f/src/components/editable.tsx:53]: Smae as the other one. Perhaps `storeKey` is less confusing.

**Review by AriaMinaei — CHANGES_REQUESTED** [review]
Requested a minor change. Other than that, good to merge!

File: packages/r3f/src/components/editable.tsx [hunk 3]
```diff
@@ -140,7 +140,7 @@ const createEditable = <Keys extends keyof JSX.IntrinsicElements>(
             visible={visible !== 'editor' && visible}
             userData={{
               __editable: true,
-              __editableName: objectKey,
+              __editableName: storeKey,
```
Comment (AriaMinaei) [inline packages/r3f/src/components/editable.tsx:143]: Probably should rename `__editableName` too, right?

Comment (AndrewPrifer) [reply packages/r3f/src/components/editable.tsx:143]: True

**Pull request merged**